### PR TITLE
test: restructure test_backend_failure test suite

### PR DESCRIPTION
**Pull request by mike-sul**
- break test_backend_failure tests into several files to improve the tests overall running time
- run python tests of each test suite concurrently

Signed-off-by: Mike Sul <ext-mykhaylo.sul@here.com>

### CODE REVIEW

File: tests/test_fixtures.py [hunk 1]
```diff
@@ -1046,3 +1046,42 @@ def func(handler_map={}):
...
+        test_run_result = test()
+        logger.info('>>> {}: {}\n'.format('OK' if test_run_result else 'FAILED', test.__name__))
+        return test_run_result
+
+    def run(self):
+        results = self._test_runner_pool.map(TestRunner.test_runner, self._tests)
+        total_result = True
+        for result in results:
+            total_result = total_result and result
+        return total_result
```
Comment (patrickvacek) [inline tests/test_fixtures.py]: Does the output from the tests running simultaneously get interleaved?

File: tests/test_customrepo_failure.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+@with_uptane_backend(start_generic_server=True)
+# TODO: Limit a number of HTTP redirects within a single request processing
+# https://saeljira.it.here.com/browse/OTA-3729
+@with_customrepo(handlers=[
+                            RedirectHandler(number_of_redirects=10, url='/primary-image.img')
+                        ])
+@with_imagerepo()
+@with_director()
+@with_aktualizr(run_mode='once', output_logs=True)
+def test_backend_failure_sanity_customrepo_update_redirect(aktualizr, uptane_repo,
```
[the remark this answers is not in the document]
Comment (mike-sul) [reply tests/test_customrepo_failure.py]: I will rename them to adjust to the test_backend_failure restructuring

File: tests/test_fixtures.py (hunk 1 above)
Comment (mike-sul) [reply tests/test_fixtures.py]: Not exactly, total_result will be False if at least one of the tests fails, it's kind of result for the overall test suite.

**Comment by mike-sul** [comment]
> Looks good, but haven't studied the details too closely yet.
> 
> The second commit implies that it is a docs-only change, but I don't see any docs-related changes. Is something missing?

I prefixed the commit message incorrectly by mistake, will update it to `test:`

**Comment by mike-sul** [comment]
@patrickvacek Updated according to the latest comments. An overall test running time is decreased by approximately 4 minutes.

`
1/6 Test #127: test_ip_secondary ................   Passed   15.03 sec
2/6 Test #132: test_update ......................   Passed   22.22 sec
3/6 Test #130: test_customrepo_failure ..........   Passed   27.81 sec
4/6 Test #131: test_treehub_failure .............   Passed   33.93 sec
5/6 Test #128: test_director_failure ............   Passed   67.12 sec
6/6 Test #129: test_imagerepo_failure ...........   Passed  167.96 sec
`

File: tests/test_fixtures.py (hunk 1 above)
Comment (patrickvacek) [reply tests/test_fixtures.py]: Sorry, I meant that I was wondering if the output printed to the screen from the tests gets mixed together from multiple tests running in parallel or if ctest or whatever is smart enough to separate them somehow.

**Comment by patrickvacek** [comment]
> An overall test running time is decreased by approximately 4 minutes.

Nice!

> 6/6 Test #129: test_imagerepo_failure ........... Passed 167.96 sec

This still seems pretty long. Obviously it's already an improvement, but do you know why this one is particularly long?

**Comment by mike-sul** [comment]
> > An overall test running time is decreased by approximately 4 minutes.
> 
> Nice!
> 
> > 6/6 Test #129: test_imagerepo_failure ........... Passed 167.96 sec
> 
> This still seems pretty long. Obviously it's already an improvement, but do you know why this one is particularly long?

because it applies the same business logic to each role's metadata which is 4 in this case.

File: tests/test_fixtures.py (hunk 1 above)
Comment (lbonn) [reply tests/test_fixtures.py]: @patrickvacek did you actually clarify that or is the question still valid?

Comment (patrickvacek) [reply tests/test_fixtures.py]: Still an open question, actually.

**Pull request merged**